api/scpca_portal/metadata_file.py

```diff
@@ -87,7 +88,26 @@ def transform_keys(data_dict: Dict, key_transforms: List[Tuple]):
     return data_dict
 
 
-def write_metadata_dicts(list_of_dicts: List[Dict], output_file_path: str, **kwargs) -> None:
+class MetadataFilenames:
+    SINGLE_CELL_METADATA_FILE_NAME = "single_cell_metadata.tsv"
+    SPATIAL_METADATA_FILE_NAME = "spatial_metadata.tsv"
+    METADATA_ONLY_FILE_NAME = "metadata.tsv"
+
+
+def get_metadata_file_name(download_config: Dict) -> str:
+    """Return metadata file name according to passed download_config."""
+    output_file_constant = (
+        f'{download_config["modality"]}_METADATA_FILE_NAME'
+        if not download_config.get("metadata_only", False)
+        else "METADATA_ONLY_FILE_NAME"
+    )
+
+    return getattr(MetadataFilenames, output_file_constant)
+
+
+def write_metadata_dicts(
+    list_of_dicts: List[Dict], output_file_path: str = None, *, buffer: io.StringIO = None, **kwargs
```

Comment:
Do you think the signature is too busy here?

Reply:
Yeah we really only need a `metadata_file.get_file_contents` that takes the list of metadata dicts and kwargs for dict writer then returns what is currently `metadata_buffer.getvalue()`. Manager the buffer internally.

If we end up needing a `metadata_file.write_metadata_dict` then we will just call `get_file_contents` internally and write it to `output_file_path`. However, I don't think we need this at this time.